PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/ValidationUtils.java

```diff
@@ -73,4 +76,31 @@ public static void checkFunctionAppName(String subscriptionId, String functionAp
             // swallow exception when get azure client
         }
     }
+
+    public static void checkResourceGroupName(String subscriptionId, String resourceGroup) {
+        if (StringUtils.isEmpty(subscriptionId)) {
+            throw new IllegalArgumentException("Subscription can not be null");
+        }
+        if (StringUtils.isEmpty(resourceGroup)) {
+            throw new IllegalArgumentException("Resource group name can not be null");
+        }
+        try {
+            final Azure azure = AuthMethodManager.getInstance().getAzureManager().getAzure(subscriptionId);
+            if (azure.resourceGroups().getByName(resourceGroup) != null) {
+                throw new IllegalArgumentException("A resource group with the same name already exists in the selected subscription.");
+            }
+        } catch (RestException e) {
+            throw new IllegalArgumentException(e.getMessage());
+        } catch (IOException e) {
+            // swallow exception when get azure client
+        }
+    }
+
+    public static void checkAppServicePlanName(String appServicePlan){
+        if (StringUtils.isEmpty(appServicePlan)) {
+            throw new IllegalArgumentException("App Service Plan name is required");
+        } else if (!appServicePlan.matches(RESOURCE_GROUP_PATTERN)) {
```

Comment:
RESOURCE_GROUP_PATTERN is only used here, should be changed to UPPER(`app_service_plan_name_path`)